Vs code golden path

BillWagner opened the pull request:
Fixes #282.

Pulled in content from @tsolarin and @cqcallaw

Getting started with VS Code.

BillWagner committed:
interim checkin to switch branches

tonerdo committed:
add Visual Studio Code "golden path" boilerplate markdown

BillWagner committed:
Finish draft

BillWagner committed:
Add sample for VS Code Golden Path

add image for VSCode debugger

Final review, and include images.

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,372 @@
+
+```js
+"options": {
+    "cwd": ${workspaceRoot}/src/app"
````
Missing `"`.

BillWagner committed:
Update Getting Started with @tsolarin's content

BillWagner committed:
respond to feedback

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,378 @@
+
+The `dotnet new` command creates a new application project. Because
+you want a library project, you're going to make a few changes to
+the generated `project.json` file.
```
This is meant for RC2/preview1, for now, right? So it can't use the new `dotnet new -t lib`?

BillWagner replied:
That's correct. It will be updated as appropriate with each release.

blackdwarf replied:
Since we are nearing the RTM and this is not merged, maybe we fix it now to use `dotnet new -t lib`?

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,378 @@
+and writes a `project.lock.json` file in the project directory. This
+file contains the full dependency tree of all the dependencies in your
+project. You don't need to read this file, its used by tools in the .NET
+SDK.
```
.NET SDK → .NET Core SDK?

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,378 @@
+Now, let's update the C# code. Let's create a `Thing` class that contains
+one public method. This method will return the sum of two numbers,
+but  will do so by converting that number to a JSON string, and then
+deserializing it. Rename the file `Program.cs` to `thing.cs`. Then, replace
```
Why is this changing the file name to lowercase?

BillWagner replied:
No real reason. Just habit when I'm in a *nx shell.

bleroy replied:
But the name of the file is the name of the class. Also, if you create a class in VS, you'll get 'Thing.cs' as the file name.

BillWagner replied:
I'll change it.

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,378 @@
+```
+
+If you are running on Windows, you'll need to update `project.json` to
+generate portable PDB files (this happens by default on Mac OSX and Linux).
````
Is it going to be clear to the reader that it's `project.json` *of the application* and not one of the other `project.json` files?

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,378 @@
+```
+
+Set a breakpoint at the `WriteLine` statement in `Main`. You do this
+by pressing the `F9` key. Open the debugger by pressing the debug
````
Isn't setting the breakpoint using mouse more intuitive?

BillWagner commented on docs/toc.md:
```diff
@@ -8,7 +8,7 @@
 ## [Setting up your integrated development environment](getting-started/devenv/index.md)
 ### [🔧 Development using Visual Studio 2015](getting-started/devenv/using-visual-studio.md)
-### [🔧 Development using Visual Studio Code](getting-started/devenv/using-visual-studio-code.md)
+### [Development using Visual Studio Code](getting-started/devenv/using-visual-studio-code.md)
```
Done. Good catch @svick (and yes, that's a feature request for docfx)

BillWagner committed:
Respond to feedback

BillWagner committed:
update case on file name.

blackdwarf commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+Prerequisites
+-------------
+
+Before starting, you'll need to install the [.NET Core SDK](https://www.microsoft.com/net/download),
```
It should be https://www.microsoft.com/net/core for best experience.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+Contributions by [Toni Solarin-Sodara](https://github.com/tsolarin)
+
+This document provides a tour of the steps and workflow to create a .NET
+Core Solution using [Visual Studio Code](http://codde.visualstudio.com).
```
codde --> code

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+This document provides a tour of the steps and workflow to create a .NET
+Core Solution using [Visual Studio Code](http://codde.visualstudio.com).
+You'll learn how to create projects, create unit tests, use the debugging
+tools, and incorproate third-party libraries via [NuGet](http://nuget.org).
```
typo incorproate

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+-------------
+
+Before starting, you'll need to install the [.NET Core SDK](https://www.microsoft.com/net/download),
+currently in a preview relese. The .NET Core SDK includes the latest release
```
typo relese
should the preview release note be removed?

BillWagner replied:
@tdykstra I fixed the typo. I think the SDK will still be a Preview with the next update.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+tests for that library project, and a console application that makes
+use of the library. You'll follow a standard folder structure for
+the three projects. Following this standard folder structure
+maeans that the .NET Core SDK tools understand the relationship between
```
typo maeans

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,381 @@
+
+```js
+{
+    projects: [
````
In Windows "projects" shows up in red with an error message saying property keys must be double-quoted.

BillWagner replied:
The code was correct (with "projects" in quotes.) UPdated.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+you want a library project, you're going to make a few changes to
+the generated `project.json` file.
+
+First, remote the `buildOptions` node:
```
remote --> remove

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,381 @@
+add the latest pre-release version of the package as a dependency:
+
+```js
+"depdendencies": {
````
typo here, and the format of the property is different from how it was entered above

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+        "version": "1.5.0-rc2-24027"
+    },
+    "Newtonsoft.Json": "9.0.1-beta1"
+}
```
if you're copying and pasting code, you need a comma at the end here

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+packages into workspace. The `dotnet restore` command updates all dependencies,
+and writes a `project.lock.json` file in the project directory. This
+file contains the full dependency tree of all the dependencies in your
+project. You don't need to read this file, its used by tools in the .NET
```
its --> it's

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,381 @@
+}
+```
+
+This makes use of a number of modern C# features that you can learn
````
A comment like this might be more helpful if it mentioned what the modern features are and linked directly to articles about them.

BillWagner replied:
Agreed. I added a brief list, but the articles haven't been written yet.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -1,16 +1,381 @@
+```
+
+After you've finished adding those dependencies, you need to install those
+packages into workspace. The `dotnet restore` command updates all dependencies,
````
The text mentions the dotnet restore command but never explicitly says to execute it now.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -1,16 +1,381 @@
+        "type": "platform",
+        "version": "1.0.0-rc2-3002702"
+    },
+    "xunit" "2.1.0",
```
missing colon

BillWagner committed:
update based on PR feedback.

svick commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+
+Set a breakpoint at the `WriteLine` statement in `Main`. You do this
+by pressing the `F9` key, or by clicking the mouse in the left margin
+on the line you want the breakpoint). 
```
Closing parenthesis without matching opening paren.

blackdwarf commented:
@BillWagner since we are nearing RTM/Preview 2 release, shouldn't these document reflect that in the commands (e.g. `dotnet new -t lib`) but also in package/metapackage versions?

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+    {
+        [Fact]
+        public void TestThing() {
+            Assert.Equal(42, new Thing().Get(19.23));
```
19.23 --> 19, 23

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+and replace the contents of the `Main` method with this line:
+
+```cs
+WriteLine($"The answer is {new Thing().Get(19, 23))}");
````
extra close paren after 19, 23

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+page.)
+
+After you install the extension, VS Code will ask that you restart the application
+to include the new extension. When you relaunch, VS Code will create a `.vscode` directory
```
When I did this, VS Code didn't create the .vscode directory when I relaunched after restarting.  It didn't create .vscode until I tried to start the debugger, then gave me an error message saying I needed to set up my environment.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+
+After you install the extension, VS Code will ask that you restart the application
+to include the new extension. When you relaunch, VS Code will create a `.vscode` directory
+with two files: `tasks.json` and `launch.json`. These two files control the debugger
```
I only got launch.json, and it has cwd under configurations/.NET Core Launch (console)

tdykstra replied:
I get tasks.json only after I click a button to configure for .NET Core.

tdykstra commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+```
+
+If you are running on Windows, you'll need to update the Application's `project.json` 
+(in the `src/app` directory) to
````
For Windows I had to do this in the library as well or I got an error message when I hit the breakpoint "WARNING: Could not load symbols for 'library.dll'. 'c:\Users\tdykstra\Source\Repos\golden\src\app\bin\Debug\netcoreapp1.0\library.pdb' is a Windows PDB. These are not supported by the cross-platform .NET Core debugger."

BillWagner committed:
two quick grammar mistakes.

cartermp commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+"dependencies": {
+    "Microsoft.NETCore.App": {
+        "type": "platform",
+        "version": "1.0.0-rc2-300702"
```
This should be `1.0.0`.

cartermp commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+```
+
+You'll also need to update the frameworks section to use 
+.NET Standards instead of `netcoreapp1.0`:
````
".NET Standards" --> ".NET Standards"

BillWagner replied:
@cartermp Did you mean ".NET Standards" -> ".NET Standard"? That's the change I'm making now.

cartermp replied:
Yep!  I seem to have misspelled my spelling suggestion :rage4:

cartermp commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+
+```js
+"frameworks" : {
+    "netstandard1.5": {
````
Should be `netstandard1.6`.

Also missing closing brace around the TFM.

cartermp commented on docs/getting-started/devenv/using-visual-studio-code.md:
````diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+```js
+"dependencies": {
+    "NETStandard.Library": {
+        "version": "1.5.0-rc2-24027"
````
In this case, there's no need to use the object notation.  Just `"NETStandard.Library":"1.6.0"` will do.

cartermp commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+    },
+    "library": {
+        "target": "project",
+        "version": "1.0.0-*"
```
We should drop the splat in here.  Even though it's fine and only a project-to-project reference, we do want to discourage the use of splat strings.

BillWagner replied:
Not sure what you mean by "drop the splat". Can you clarify?

mairaw replied:
I'm guessing it's the *.

cartermp replied:
Yup, that's what I'm referring to.

cartermp commented:
LGTM modulo minor feedback.  /cc @bleroy

bleroy commented on docs/getting-started/devenv/using-visual-studio-code.md:
```diff
@@ -12,19 +12,387 @@ ms.devlang: dotnet
+The source for this tutorial is available on
+[GitHub](https://github.com/dotnet/core-docs/tree/master/samples/getting-started/golden).
+
+Start Visual Studio Code. Press Ctrl + '`' (the back-quote character) to open
```
Escape that backtick.

BillWagner committed:
Updates on Feedback.

At this point it should be ready, short of updating versions to RMT.

BillWagner committed:
catch a couple proofreading errors.

BillWagner commented:
@blackdwarf I'm updating to use `dotnet new -t lib`. I'm thinking I should update for `dotnet new -t xunittest` as well. Correct?

blackdwarf commented:
@BillWagner yes please. This way, we get the eyeballs on these new features (that I added wink-wink :)) so that people can use them in day-to-day work.

BillWagner committed:
Update to use the Preview 2 tooling and RTM libs

This made the topic somewhat shorter, I'd like a quick review
@blackdwarf to make sure this is good to go.

BillWagner commented:
@blackdwarf Just merged all those updates to this PR. Can you give it a look and make sure I described those new features well.

I was really happy to remove all the extra steps that weren't needed any more. It's a much better overall experience.

blackdwarf commented:
LGTM! :shipit:

cartermp commented:
:shipit: :shipit: :shipit: :shipit: :shipit:

BillWagner commented:
:thumbsup: :fireworks:
